### PR TITLE
Basic functional profiling 

### PR DESCRIPTION
**Pull request by jamiebramwell**
- Adds new basic `Functional` benchmark tests using Caliper
- Corrects the existing `HeatTransfer` benchmark tests to use the new APIs.
- Adds a Ubuntu22 dependency to avoid over-Spackification when building with profiling tools.

### CODE REVIEW

**Comment by jamiebramwell** [comment]
Sample output:

```
Path                    Min time/rank Max time/rank Avg time/rank Time %    
scalar H1                    0.000019      0.000019      0.000019  0.000003 
  dimension 2, order 1       0.041346      0.041346      0.041346  0.005765 
    residual evaluation      0.014610      0.014610      0.014610  0.002037 
    compute gradient         0.018494      0.018494      0.018494  0.002579 
    apply gradient           0.006283      0.006283      0.006283  0.000876 
    assemble gradient        0.026324      0.026324      0.026324  0.003670 
  dimension 2, order 2       0.239186      0.239186      0.239186  0.033351 
    residual evaluation      0.023415      0.023415      0.023415  0.003265 
    compute gradient         0.041711      0.041711      0.041711  0.005816 
    apply gradient           0.016217      0.016217      0.016217  0.002261 
    assemble gradient        0.166346      0.166346      0.166346  0.023194 
  dimension 3, order 1       1.162226      1.162226      1.162226  0.162055 
    residual evaluation      0.280935      0.280935      0.280935  0.039172 
    compute gradient         0.511811      0.511811      0.511811  0.071364 
    apply gradient           0.146481      0.146481      0.146481  0.020425 
    assemble gradient        1.039805      1.039805      1.039805  0.144985 
  dimension 3, order 2      19.810753     19.810753     19.810753  2.762308 
    residual evaluation      1.003750      1.003750      1.003750  0.139958 
    compute gradient         1.810960      1.810960      1.810960  0.252511 
    apply gradient           0.601789      0.601789      0.601789  0.083910 
    assemble gradient       16.706101     16.706101     16.706101  2.329412 
vector H1                    0.000018      0.000018      0.000018  0.000003 
  dimension 2, order 1       0.133662      0.133662      0.133662  0.018637 
    residual evaluation      0.016906      0.016906      0.016906  0.002357 
    compute gradient         0.048992      0.048992      0.048992  0.006831 
    apply gradient           0.015156      0.015156      0.015156  0.002113 
    assemble gradient        0.106378      0.106378      0.106378  0.014833 
  dimension 2, order 2       1.170470      1.170470      1.170470  0.163204 
    residual evaluation      0.042799      0.042799      0.042799  0.005968 
    compute gradient         0.124428      0.124428      0.124428  0.017350 
    apply gradient           0.038155      0.038155      0.038155  0.005320 
    assemble gradient        0.723329      0.723329      0.723329  0.100857 
  dimension 3, order 1      10.100768     10.100768     10.100768  1.408399 
    residual evaluation      0.658080      0.658080      0.658080  0.091759 
    compute gradient         2.309018      2.309018      2.309018  0.321957 
    apply gradient           0.589190      0.589190      0.589190  0.082154 
    assemble gradient       11.024530     11.024530     11.024530  1.537203 
  dimension 3, order 2     349.291853    349.291853    349.291853 48.703438 
    residual evaluation      2.508675      2.508675      2.508675  0.349797 
    compute gradient         8.382576      8.382576      8.382576  1.168823 
    apply gradient           2.388804      2.388804      2.388804  0.333082 
    assemble gradient      283.838558    283.838558    283.838558 39.576971 
```

**Review by johnbowen42 — APPROVED** [review]
LGTM

File: src/serac/physics/benchmarks/benchmark_functional.cpp [hunk 1]
```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2019-2023, Lawrence Livermore National Security, LLC and
```
Comment (white238) [inline src/serac/physics/benchmarks/benchmark_functional.cpp]:
```suggestion
// Copyright (c) 2019-2024, Lawrence Livermore National Security, LLC and
```

**Review by white238 — APPROVED** [review]

File: src/docs/sphinx/quickstart.rst [hunk 2]
```diff
@@ -243,7 +243,7 @@ install cmake, MPICH, openblas, OpenGL, and the various developer tools using th
 
    $ sudo apt-get update
    $ sudo apt-get upgrade
-   $ sudo apt-get install cmake libopenblas-dev libopenblas-base mpich mesa-common-dev libglu1-mesa-dev freeglut3-dev cppcheck doxygen libreadline-dev python3-sphinx python3-pip clang-format-10 m4
+   $ sudo apt-get install cmake libopenblas-dev libopenblas-base mpich mesa-common-dev libglu1-mesa-dev freeglut3-dev cppcheck doxygen libreadline-dev python3-sphinx python3-pip clang-format-10 m4 elfutils
```
Comment (chapman39) [inline src/docs/sphinx/quickstart.rst:246]: this part of quickstart is out of date. should the addition of `elfutils` be added to the dockerfiles in `scripts/docker`?

**Pull request merged**